integration_tests/transaction_stress_test.go

```diff
@@ -0,0 +1,147 @@
+			valAddr, err := validator.keyInfo.GetAddress()
+			s.Require().NoError(err)
+			for j := 0; j < int(transactionsPerValidator); j++ {
+				s.Require().NoError(s.SendEthTransaction(&validator.ethereumKey, gravityContract, PackSendToCosmos(testERC20contract, valAddr, sendAmt)))
+			}
+
+			s.T().Logf("%d Tx sent.", transactionsPerValidator)
+		}
+
+		var gravityDenom string
+		for i, validator := range s.chain.validators {
+			s.Require().Eventuallyf(func() bool {
+				s.T().Logf("Checking validator %d", i+1)
+
+				kb, err := validator.keyring()
+				s.Require().NoError(err)
+				valAddr, err := validator.keyInfo.GetAddress()
+				s.Require().NoError(err)
+				clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &kb, "val", valAddr)
+				s.Require().NoError(err)
+
+				bankQueryClient := banktypes.NewQueryClient(clientCtx)
+				res, err := bankQueryClient.AllBalances(context.Background(),
+					&banktypes.QueryAllBalancesRequest{
+						Address: valAddr.String(),
+					})
+				if err != nil {
+					s.T().Logf("error: %s", err)
+					return false
+				}
+
+				gbQueryClient := types.NewQueryClient(clientCtx)
+				denomRes, err := gbQueryClient.ERC20ToDenom(context.Background(),
+					&types.ERC20ToDenomRequest{
+						Erc20: testERC20contract.String(),
+					})
+				if err != nil {
+					s.T().Logf("error querying ERC20 denom %s, %e", testERC20contract.String(), err)
+					return false
+				}
+				s.Require().False(denomRes.CosmosOriginated, "ERC20-originated denom marked as cosmos originated")
+				gravityDenom = denomRes.Denom
+
+				for _, coin := range res.Balances {
+					if coin.Denom == gravityDenom && coin.Amount.Equal(sdk.NewInt(cosmosSentAmt*transactionsPerValidator)) {
+						s.T().Logf("Expected funds received for validator %d, balance: %v", i+1, coin)
+						return true
+					}
+				}
+
+				s.T().Logf("balance not found, received %v", res.Balances)
+
+				return false
+			}, 105*time.Second, 10*time.Second, "balance never found on cosmos")
+		}
+		fmt.Println("Ethereum -> Cosmos stress test completed.")
+
+		for i, validator := range s.chain.validators {
+			s.T().Logf("sending %d tx's to ethereum for validator %d ..", transactionsPerValidator, i+1)
+
+			keyring, err := validator.keyring()
+			s.Require().NoError(err)
+			valAddr, err := validator.keyInfo.GetAddress()
+			s.Require().NoError(err)
+			sendToEthereumMsg := types.NewMsgSendToEthereum(
+				valAddr,
+				validator.ethereumKey.address,
+				sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(ethSentAmt)},
+				sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(0)},
+			)
+
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &keyring, "val", valAddr)
+			s.Require().NoError(err)
+
+			for j := 0; j < int(transactionsPerValidator); j++ {
+				response, err := s.chain.sendMsgs(*clientCtx, sendToEthereumMsg)
+				if err != nil {
+					s.T().Logf("error: %s", err)
+				} else {
+					if response.Code != 0 {
+						if response.Code != 32 {
```

Comment:
sorry but could I ask what does it mean for `response.code` being 32?

Reply:
its code for wrong account sequence, but I end up removing that part because we are enforcing strict check balance at the end so the transaction should not fail